lms/djangoapps/completion/tests/test_models.py

```diff
@@ -41,6 +39,27 @@ def setUp(self):
             completion=0.5,
         )
 
+
+class SubmitCompletionTestCase(CompletionSetUpMixin, TestCase):
+    """
+    Test that BlockCompletion.objects.submit_completion has the desired
+    semantics.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(SubmitCompletionTestCase, cls).setUpClass()
+        cls.overrider = waffle.waffle().override(waffle.ENABLE_COMPLETION_TRACKING, True)
+        cls.overrider.__enter__()
```

Comment:
Too bad this doesn't work as a class decorator.

Reply:
I know. I tried. For what it's worth, it does work as a (function) decorator in versions of python after 3.2. Even then I don't think it would work on the life cycle of a class.

Reply:
If you do go down the Waffle Flag route, I'm not sure if there would yet be a simple way to do this.